Automattic/jetpack
# Premium Content block: Don't allow block to be nested

Since we currently don't allow for blocks to be transformed to a Premium Content from inside of an existing Premium content, we'll also block the possibility to add new block. This restriction happens only for direct children of the block.

However, a user can still add a premium content block if they use an intermediary block (e.g. Group). This behaviour was not restricted due to technical shortcomings of the current Gutenberg API.

While this is not an ideal solution, it partially handles the restriction and avoids the block navigation issues - the navigation issue is not reproducible when the block is added from an intermediary block.





Fixes #22619

#### Changes proposed in this Pull Request:

* Use allowedBlocks in subscriber view and guest view to filter the premium-content/container block.
#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Use Jetpack Beta and switch to this branch on your Atomic site
* Apply D76012-code on your sandbox
* Create a new post and add a Premium content block
* Go to the subscriber view and try to add a Premium content block again.
* Notice that "Premium Content" block is not available in the list.
* Repeat the test for "Guest" view. 
* In the subscriber view, add a "Group" block
* Add Premium content inside the group, you'll notice that's possible.
* Make sure that the block navigation hierarchy is correct.
